File: src/main/java/projectir4/elearning/model/Student.java
package projectir4.elearning.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;

import java.util.List;

@Entity
public class Student {


    @Id
    @GeneratedValue
    private long id;
    private String firstname;
    private String lastname;
    private String email;
    private String telephone;


    //manage and not ignore so that when calling we get to see the enrollments only
    @JsonIgnoreProperties("student")
    @OneToMany(mappedBy = "student", fetch = FetchType.EAGER, cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Enrollment> enrollments;


    public long getId(){ return id; }

    public void setId(long id){ this.id = id; }

    public String getFirstname() {return firstname;}
    public void setFirstname(String firstname) {this.firstname = firstname;}

    public String getLastname() {return lastname;}

    public void setLastname(String lastname) {this.lastname = lastname;}

    public String getEmail() {return email;}

    public void setEmail(String email) {this.email = email;}

    public String getTelephone() {return telephone;}

    public void setTelephone(String telephone) {this.telephone = telephone;}

    public List<Enrollment> getEnrollments() {return enrollments;}

    public void setEnrollments(List<Enrollment> enrollments) {this.enrollments = enrollments;}
}

File: Graphics/src/app/components/students/student.ts
import {Enrollment} from '../enrollments/enrollment';

export class Student {

  id?: number;
  firstname: string;
  lastname: string;
  email: string;
  telephone: string;
  enrollments?: Enrollment[];

  constructor(firstname: string, lastname: string, email: string, telephone: string, enrollments: Enrollment[]) {
    this.firstname = firstname;
    this.lastname = lastname;
    this.email = email;
    this.telephone = telephone;
    this.enrollments = enrollments;
  }
}
